src/main/java/org/kiwiproject/dropwizard/util/job/MonitoredJobs.java

```diff
@@ -0,0 +1,156 @@
+     * @param name          the name of the job
+     * @param schedule      the schedule for the job
+     * @param runnable      the task that will be run inside the Monitored Job
+     * @param decisionFn    the function that will decide of the job should run
+     * @param executor      the scheduled executor to use to schedule the job
+     * @return the new {@link MonitoredJob}
+     */
+    public static MonitoredJob registerJob(Environment env,
+                                           String name,
+                                           JobSchedule schedule,
+                                           Runnable runnable,
+                                           Function<MonitoredJob, Boolean> decisionFn,
+                                           ScheduledExecutorService executor) {
+        validateJob(name, schedule);
+
+        var job = MonitoredJob.builder()
+                .name(name)
+                .task(runnable)
+                .decisionFunction(decisionFn)
+                .build();
+
+        var healthCheck = registerHealthCheck(env, name, schedule, job);
+
+        scheduleJob(executor, schedule, job);
+
+        JOBS.put(name, new JobContext(job, healthCheck, executor));
+        return job;
+    }
+
+    private static void validateJob(String name, JobSchedule schedule) {
+        checkArgument(isNull(JOBS.get(name)), IllegalArgumentException.class,
+                "Jobs cannot be registered more than once with the same name: {}", name);
+
+        checkArgumentNotNull(schedule.getInitialDelay(),
+                "Job '{}' must specify a non-null initial delay", name);
+
+        checkArgumentNotNull(schedule.getIntervalDelay(),
+                "Job '{}' must specify a non-null interval delay", name);
+    }
+
+    private static MonitoredJobHealthCheck registerHealthCheck(Environment env, String name,
+                                                               JobSchedule schedule, MonitoredJob job) {
+
+        LOG.debug("Creating and registering health check for job: {}", name);
+
+        var healthCheck = MonitoredJobHealthCheck.builder()
+                .job(job)
+                .expectedFrequency(schedule.getIntervalDelay())
+                .build();
+
+        env.healthChecks().register(f("Job: {}", name), healthCheck);
+        return healthCheck;
+    }
+
+    private static void scheduleJob(ScheduledExecutorService executor, JobSchedule schedule, MonitoredJob job) {
+        LOG.debug("Scheduling job: {} to run every: {}", job.getName(), schedule.getIntervalDelay());
+        executor.scheduleWithFixedDelay(job, schedule.getInitialDelay().toSeconds(),
+                schedule.getIntervalDelay().toSeconds(), TimeUnit.SECONDS);
+    }
+
+    /**
+     * Container that holds the newly created {@link MonitoredJob}, {@link MonitoredJobHealthCheck}
+     * and the {@link Executor} running the job.
+     */
+    @Getter
+    @AllArgsConstructor
+    public static class JobContext {
```

Comment:
Who uses this? Actually, how does someone get one? All the above methods return `MonitoredJob`

Reply:
Hmm... good point. There wasn't anything in the original code that exposed this.  Or did anything with it.

Reply:
Should I just remove all of that complexity (memory usage)?

Reply:
It is only used with the concurrent hash map to track if the same job name is used.  I could easily just make that a set.

Reply:
That is what I did.